### ISSUE TITLE
JiT meets Residual Layers

### ISSUE THREAD

**Issue by yihua7** [LTH14/JiT#8]
Hi,  

Thanks for the great work!  

The conclusion in the paper is clear and convincing. However, I am curious about the performance of predicting different attributes (e.g., **x₀, velocity (v), and noise**) with respect to different model architectures.  
For instance:  
- If the model is a **UNet** with a long residual layer spanning from the beginning to the end, the prediction of **v** can be easily handled with a final linear layer.  
- Similarly, models with **residual blocks** may behave differently in predicting these attributes.  

I noticed that **JiT** does not apply such residual blocks. My question is:  
- Does the advantage of predicting **x₀** over **v** still hold for models with architectures like **UNet** or models incorporating **residual transformer blocks**?  

Looking forward to your insights!

**Comment by LTH14** [comment]
Thanks for your interest! Regarding UNet, you can see this [issue](https://github.com/LTH14/JiT/issues/2#issuecomment-3549442668).

Regarding residual blocks: in fact, considering the transformation from x to v: v = (x_pred - z) / (1 - t), this is an explicit residual connection from the input z to the output, up to a scaling factor -1 / (1 - t). If other neural network architectures also have this direct t-dependent residual connection from the input to the output (or a skip connection in the network where there is no information bottleneck), I think it could implicitly learn such a transformation.

**Comment by yihua7** [comment]
Thanks for the answer! A linear out layer modulated by t with a jumping connection from x_t may also achieve similar results.

**Comment by blairstar** [comment]
#### In fact, when optimizing the prediction of $x_0$, we are also implicitly optimizing another form of predicting $\varepsilon - x_0$; see the derivation below. ####

***

Suppose there is a neural network function $g_\theta(x_t, t)$:

$$
g_\theta(x_t, t) \xrightarrow{\text{fit}} x_0 \qquad x_t = t \cdot \epsilon + (1 - t) \cdot x_0
$$

The method of constructing $x_t$ above allows us to derive the following relationship:

$$
\epsilon - x_0 = \frac{x_t - x_0}{t}
$$

Therefore, we can construct a new function to predict $\epsilon - x_0$:

$$
f_\theta(x_t, t) = \frac{ x_t - g_\theta(x_t, t) }{ t }   \qquad (1)
$$

It can be seen that the new function $f_\theta(x_t, t)$ introduces no new optimizable parameters, and the input is exactly the same as $g_\theta(x_t, t)$, which consists of $x_t$ and $t$.

Next, we analyze the objective function to be optimized.

Suppose there is a training data point $x'_0$, and we optimize the two functions respectively.

For $g_\theta(x_t, t)$, our objective function is to minimize:

$$
\min_\theta \Vert  g_\theta(x'_t, t) - x'_0 \Vert ^2 \qquad x'_t = t \cdot \epsilon + (1 - t) \cdot x_0
$$

For $f_\theta(x_t, t)$, our objective function is to minimize:

$$
\min_\theta \Vert  f_\theta(x'_t, t) - (\epsilon - x'_0) \Vert ^2 \qquad x'_t = t \cdot \epsilon + (1 - t) \cdot x_0
$$

The objective function regarding $f_\theta(x_t, t)$ can be transformed into another form.

$$
\begin{aligned}
\Vert f_\theta(x'_t, t) - (\epsilon - x'_0)\Vert^2 &= \left\Vert \frac{x'_t - g_\theta(x'_t, t)}{t} - (\epsilon - x'_0) \right\Vert^2 \\
&= \left\Vert \frac{x'_t - g_\theta(x'_t, t) - t \cdot (\epsilon - x'_0)}{t} \right\Vert^2 \\
&= \left\Vert \frac{x'_0 - g_\theta(x'_t, t)}{t} \right\Vert^2 \\
&= \frac{1}{t^2} \left\Vert x'_0 - g_\theta(x'_t, t) \right\Vert^2
\end{aligned}
$$

Therefore, we have the following relationship:

$$
\Vert g_\theta(x'_t, t) - x'_0 \Vert^2 = t^2 \cdot \Vert f_\theta(x'_t, t) - (\epsilon - x'_0) \Vert^2
$$

So, it can be understood this way: When **[using $g_\theta(x_t, t)$ to fit $x_0$]**, it also implicitly implies **[using $f_\theta(x_t, t)$ to fit $\epsilon - x_0$]**, except that a weight coefficient $t^2$ is multiplied to the objective function of $f_\theta(x_t, t)$.

Therefore, we can consider that **the optimization processes of different prediction forms exist simultaneously.**

If the performance of **[using $g_\theta(x_t, t)$ to fit $x_0$]** is better than **[using $g_\theta(x_t, t)$ to fit $\epsilon - x_0$]**, it is equivalent to having a pair of control experiments—Experiment 1 and Experiment 2—and the result of Experiment 1 is better than that of Experiment 2:

**Experiment 1:**

$$
\min_\theta \ t^2 \cdot \Vert f_\theta(x'_t, t) - (\epsilon - x'_0) \Vert^2
$$

**Experiment 2:**

$$
\min_\theta \Vert g_\theta(x'_t, t) - (\epsilon - x'_0) \Vert^2
$$

**Conclusion:** Therefore, we can  infer that: the network structure of $g_\theta(x_t, t)$ is worse than that of $f_\theta(x_t, t)$ (note that the difference between the two is very small, see Equation 1), or the weights for different times $t$ are not configured appropriately.

Additionally, we know that $t$ is an input value and is a constant. If the model corresponding to each $t$ is trained and optimized separately, then there is no need to consider the weight coefficient.
